### PR TITLE
server: check kill flag for load data

### PR DESCRIPTION
**Pull request by cfzjywxk**
### What problem does this PR solve? 

- Let the `load data` statement could be killed by `kill tidb query number`
- Add draining data logic to avoid reset the connection  encountering an error

### What is changed and how it works?

Check the ``kill` flag after every commit batch

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
1. start load statement
2. execute kill tidb query in another session

Code changes

Side effects

Related changes

Release note

### CODE REVIEW

File: server/conn.go [hunk 1]
```diff
@@ -1128,6 +1128,18 @@ func (cc *clientConn) handleLoadData(ctx context.Context, loadDataInfo *executor
 	// processStream process input data, enqueue commit task
 	go processStream(ctx, cc, loadDataInfo)
 	err = loadDataInfo.CommitWork(ctx)
+	if terror.ErrorEqual(err, executor.ErrQueryInterrupted) {
+		// drain the data from client conn util empty packet received, otherwise the connection will be reset
```
Comment (cfzjywxk) [inline server/conn.go]: Or should we just let the `kill` finish and connection be reset, it may take lots of time reading a large file
@jackysp @imtbkcat

Comment (jackysp) [reply server/conn.go]: What does MySQL do?

Comment (cfzjywxk) [reply server/conn.go]: For MySQL, when killed is received, the processing thread will read all the reamin data, and then report error
```
    if (read_file_from_client)
      read_info.skip_data_till_eof();
```
@jackysp @imtbkcat

Comment (jackysp) [reply server/conn.go]: Got.

**Comment by cfzjywxk** [comment]
I've changed the data draining logic for all non nil errors instead of `ErrInterrupted`,  this will change error to be reported but not kill connection
```
before
mysql> load data local   infile  "/data/dat"  into table lt;        
ERROR 2013 (HY000): Lost connection to MySQL server during query

after
mysql> load data local   infile  "/data/dat"  into table lt;
ERROR 1105 (HY000): test error
mysql> select 1+1;
+-----+
| 1+1 |
+-----+
|   2 |
+-----+
1 row in set (0.00 sec)

```

@imtbkcat @jackysp PTAL

**Pull request merged**